pages/learn/electives.tsx

```diff
@@ -132,7 +131,14 @@ const LearnElectivesPage: CustomPage<{
                 }
                 enrolledElective={enrolledElective}
                 disabled={
-                  !inEnrollmentPeriod || enrolledElective?.id === selectedID
+                  // Check if it's in Enrollment Period.
+                  !inEnrollmentPeriod ||
+                  // Check if it's a previously selected subject.
+                  previouslyEnrolled.includes(selectedID!) ||
+                  // Check if it's a full class subject or not.
+                  (selectedDetail === null
+                    ? false
+                    : selectedDetail!.cap_size <= selectedDetail!.class_size)
```

Comment:
This part isn’t necessary as these checks already exist inside Choose Button.

https://github.com/suankularb-wittayalai-school/mysk-frontend/blob/66a6cd928b07c58f6d756e2852755765fa0e6258/components/elective/ChooseButton.tsx#L48-L54

Reply:
Apparently, it does not work then when I'm testing the feature, sowwy :(.

Reply:
If you read the code, it seems rather obvious that ChooseButton does not have cap/class size logic, but it does check for other things.